### PR TITLE
Rpc speedup

### PR DESCRIPTION
**Pull request by wburkhardt**
speed: greatly improved speed, mainly for bigger NC messages
add: mode 1.1 protocol error handling (raise NetconfProtocolError)
fix: python 2.7 socket attribute error
test: more unit-tests

### CODE REVIEW

File: netconf_client/session.py [hunk 1]
```diff
@@ -108,8 +109,10 @@ def _recv_loop(self):
         while True:
             try:
                 msg = self.parser.send(self.mode)
+            except socket_error:
```
[the remark this answers is not in the document]
Comment (wburkhardt) [reply netconf_client/session.py:112]: Hmm, my assumption was that a socket exception might happen if the connection has been closed, all data was successfully read, and then the parser wants to read some more data, which is not an error to be logged in my mind.  Maybe, I'm wrong and this special case cannot happen, so this part should be removed again.

[the remark this answers is not in the document]
Comment (Nitori-) [reply netconf_client/session.py:112]: I do think that it's possible, but in that case I think it's OK to log the message, it's just at the `info` level and may provide useful information.

File: test/test_parse.py [hunk 2]
```diff
@@ -15,30 +22,280 @@ def recv(self, _=-1):
...
 
 
 def test_multiple_messages_single_chunk_10():
     g = parse_messages(MockStream([b"Foo]]>]]>", b"Bar]]>]]>"]), "1.0")
     assert next(g) == b"Foo"
     assert next(g) == b"Bar"
+    with pytest.raises(EndOfStream):
+        next(g)
+
+
```
[the remark this answers is not in the document]
Comment (wburkhardt) [reply test/test_parse.py:43]: Yes, basically a good idea. I have already considered this, but found out that it does not really improve the readability (for my eyes). It only works if the body always has the content (which is true for some of the test functions).
In a next version (if any), and if there are more test functions, we (I?) will actually use fixtures (pytest.mark.parametrize).

File: netconf_client/session.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (wburkhardt) [reply netconf_client/session.py:112]: Ok - I revert my change and send a new commit...

**Pull request merged**